x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -232,19 +238,134 @@ private static void validateField(Map<?, ?> properties, String fieldName, boolea
         }
     }
 
-    private XContentBuilder resolveEnrichMapping(final EnrichPolicy policy) {
-        // Currently the only supported policy type is EnrichPolicy.MATCH_TYPE, which is a keyword type
-        final String keyType;
-        final CheckedFunction<XContentBuilder, XContentBuilder, IOException> matchFieldMapping;
+    private void resolveEnrichMapping(final EnrichPolicy policy, ActionListener<XContentBuilder> resultListener) {
         if (EnrichPolicy.MATCH_TYPE.equals(policy.getType())) {
-            matchFieldMapping = (builder) -> builder.field("type", "keyword").field("doc_values", false);
-            // No need to also configure index_options, because keyword type defaults to 'docs'.
+            createEnrichMappingBuilder((builder) -> builder.field("type", "keyword").field("doc_values", false), resultListener);
+        } else if (EnrichPolicy.RANGE_TYPE.equals(policy.getType())) {
+            resolveRangeTypeEnrichMapping(policy, resultListener);
         } else if (EnrichPolicy.GEO_MATCH_TYPE.equals(policy.getType())) {
-            matchFieldMapping = (builder) -> builder.field("type", "geo_shape");
+            createEnrichMappingBuilder((builder) -> builder.field("type", "geo_shape"), resultListener);
         } else {
-            throw new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType());
+            resultListener.onFailure(new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType()));
         }
+    }
+
+    private void resolveRangeTypeEnrichMapping(EnrichPolicy policy, ActionListener<XContentBuilder> resultListener) {
+        GetFieldMappingsRequest fieldMappingsRequest = new GetFieldMappingsRequest().indices(
+            policy.getIndices().stream().toArray(String[]::new)
+        ).fields(policy.getMatchField());
+
+        client.execute(GetFieldMappingsAction.INSTANCE, fieldMappingsRequest, listener.delegateFailure((l, response) -> {
```

Comment:
Instead of invoking the get field mapping action, can range type (with a potential date format) be resolved from the response of the get index request that happened a few steps earlier?

Reply:
good catch, I changed it so it reuses the index request see https://github.com/elastic/elasticsearch/pull/76110/commits/910db97ce4ddf0dace3c81415d9ceabc42bd3062

Interesting how the java object structure is different different and, of course, that we add logic at all the call sites instead of some centralized util or convenience methods on the response object itself.  Passing this on to future us.

Reply:
iirc the get index api returns the metadata as is stored in cluster state. Whereis get mappings api has a different response structure for consuming the response in Kibana.